cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -164,16 +169,25 @@ void publishToControlChannel(TetheringControlMessage message) throws Exception {
   }
 
   /**
-   * Use GZIPOutputStream to compress the LocalFile
+   * Use GZIPOutputStream to compress the LocalFile. First attempts to read from location factory; if the file does
+   * not exist in the location factory base path, attempts to read from local disk.
    */
   @VisibleForTesting
   byte[] getLocalFileAsCompressedBytes(LocalFile localFile) throws IOException {
-    File file = new File(localFile.getURI());
+    LOG.trace("Compressing local file with name '{}' and URI '{}'", localFile.getName(), localFile.getURI());
+    InputStream in;
+    try {
+      in = locationFactory.create(localFile.getURI()).getInputStream();
+    } catch (IllegalArgumentException e) {
```

Comment:
This makes it dependent on the underlying implementation of the LocationFactory throwing IllegalArgumentException.

it's cleaner to do something like get the locationFactory.getHomeLocation() once and get the scheme from it, then compare it to the localFile URI scheme. Otherwise a change in the bound LocationFactory or implementation can change this behavior as well.

Reply:
Fixed